gcc/d/gdc.texi

```diff
@@ -120,196 +120,662 @@ and the Info entries for @file{gdc} and @file{gcc}.
 
 @c man begin DESCRIPTION gdc
 
-The @command{gdc} command is a frontend to @command{gcc} and
-supports many of the same options.  @xref{Option Summary, , Option
-Summary, gcc, Using the GNU Compiler Collection (GCC)}.  This manual
-only documents the options specific to @command{gdc}.
-Most of these have both positive and negative forms; the negative
-form of -ffoo would be -fno-foo.
-This manual documents only one of these two forms, whichever one is
-not the default.
+The @command{gdc} command is a frontend to @command{gcc} and supports many
+of the same options.  @xref{Option Summary, , Option Summary, gcc,
+Using the GNU Compiler Collection (GCC)}.  This manual only documents
+the options specific to @command{gdc}.
 
-The @command{gdc} command may be used to compile D source code into
-an object file, link a collection of object files together, or do both
-in sequence.
+@c man end
 
-Only the most useful options are listed here; see below for the
-remainder.
+@menu
+* Input and Output files::  Controlling the kind of output:
+                            an executable, object files, assembler files,
+* Runtime Options::         Options controlling runtime behaviour
+* Directory Options::       Where to find module files
+* Code Generation::         Options controlling the output of gdc
+* Warnings::                Options controlling warnings specific to gdc
+* Linking::                 Options influceing the linking step
+* Developer Options::       Options you won't use
+* Environment Variables::   Environment variables that affect @command{gdc}.
+@end menu
 
-@c man end
+@c man begin OPTIONS
 
-@c man begin OPTIONS gdc
+@node Input and Output files
+@section Input and Output files
+@cindex suffixes for D source
+@cindex D source file suffixes
 
-@table @gcctabopt
-@item -frelease
-@cindex @option{-frelease}
-Compile release version.
+For any given input file, the file name suffix determines what kind of
+compilation is done.  The following kinds of input file names are supported:
 
-@item -funittest
-@cindex @option{-funittest}
-Compile in unittest code.
+@table @gcctabopt
+@item @var{file}.d
+D source files.
+@item @var{file}.dd
+Ddoc source files.
+@item @var{file}.di
+D interface files.
+@item @var{file}.o
+Object files to link in.
+@item @@@var{file}
+A file containing a whitespace-separated list of input file names.
+Each named file is compiled, just as if it had been on the command line.
```

Comment:
Yes, this particular feature is handled by gcc itself, not us - it's just a small copy-pasto just so there is a point of reference.

From gcc manpage:
```
The file may itself contain additional @file options; any such options will be processed recursively.
```
This can be included here also.